File: src/main/java/dev/lbuddyboy/flash/user/grant/comparator/GrantWeightComparator.java
package dev.lbuddyboy.flash.user.grant.comparator;

import dev.lbuddyboy.flash.rank.comparator.RankWeightComparator;
import dev.lbuddyboy.flash.user.model.Grant;

import java.util.Comparator;

public class GrantWeightComparator implements Comparator<Grant> {

    @Override
    public int compare(Grant grant, Grant otherGrant) {
        return new RankWeightComparator().compare(grant.getRank(), otherGrant.getRank());
    }
}

File: src/main/java/dev/lbuddyboy/flash/user/comparator/GrantWeightComparator.kt
package dev.lbuddyboy.flash.user.comparator

import dev.lbuddyboy.flash.rank.comparator.RankWeightComparator
import dev.lbuddyboy.flash.user.model.Grant

class GrantWeightComparator : Comparator<Grant> {
    override fun compare(grant: Grant, otherGrant: Grant): Int {
        return RankWeightComparator().compare(grant.rank, otherGrant.rank)
    }
}